### PR TITLE
chore: remove IAM/RM related GH actions in favour of Source.

### PR DESCRIPTION
**Pull request by sashaostrikov**
Test plan:
Try new actions as soon as this PR is merged.

### CODE REVIEW

File: .github/workflows/label-move.yml [hunk 1]
```diff
@@ -274,23 +247,23 @@ jobs:
...
-  repo-management-board:
+  source-board:
     runs-on: ubuntu-latest
     env:
       PROJECT_ID: PVT_kwDOADy5QM0Pnw # https://github.com/orgs/sourcegraph/projects/209
       GITHUB_TOKEN: ${{ secrets.GH_PROJECTS_ACTION_TOKEN }}
     steps:
       - name: Get issue if relevant
-        if: ${{ contains(github.event.issue.labels.*.name, 'team/repo-management') }}
+        if: ${{ contains(github.event.issue.labels.*.name, 'team/source') }}
```
Comment (mrnugget) [inline .github/workflows/label-move.yml:257]: Would be nice if we could keep checking for `team/iam` and `team/repo-management` for backwards-compatible reasons.

File: .github/workflows/label-notify.yml [hunk 2]
```diff
@@ -22,6 +22,5 @@ jobs:
                   team/delivery=@sourcegraph/delivery
                   team/security=@dcomas
                   team/dev-experience=@jhchabran @kstretch9
-                  team/repo-management=@ryphil @mrnugget
                   team/devops=@sourcegraph/cloud-devops
-                  team/iam=@sourcegraph/iam
```
Comment (mrnugget) [inline .github/workflows/label-notify.yml:27]: Can we leave those in and also tag `@sourcegraph/source`? Then in 2-3 months we can remove this.

File: .github/workflows/move-labled-issues-to-project.yml [hunk 3]
```diff
@@ -20,12 +20,12 @@ jobs:
           labeled: team/cloud
           label-operator: OR
 
-      - name: Move team/repo-management issues
+      - name: Move team/source issues
         uses: actions/add-to-project@main
         with:
           project-url: https://github.com/orgs/sourcegraph/projects/209
```
Comment (mrnugget) [inline .github/workflows/move-labled-issues-to-project.yml:26]: URL is still the same, yes?

**Review by mrnugget — CHANGES_REQUESTED** [review]
Sorrynotsorry, requesting changes :D

File: .github/workflows/move-labled-issues-to-project.yml (hunk 3 above)
Comment (sashaostrikov) [reply .github/workflows/move-labled-issues-to-project.yml:26]: yep, the RM board was renamed to Source

**Comment by sashaostrikov** [comment]
@mrnugget makes total sense, added the labels back, created an issue to remove them after 2-3 months

File: .github/workflows/label-move.yml [hunk 4]
```diff
@@ -274,23 +247,23 @@ jobs:
...
-  repo-management-board:
+  source-board:
     runs-on: ubuntu-latest
     env:
       PROJECT_ID: PVT_kwDOADy5QM0Pnw # https://github.com/orgs/sourcegraph/projects/209
       GITHUB_TOKEN: ${{ secrets.GH_PROJECTS_ACTION_TOKEN }}
     steps:
-      - name: Get issue if relevant
-        if: ${{ contains(github.event.issue.labels.*.name, 'team/repo-management') }}
+      - name: Get issue if relevant # checking for "source", "iam" and "repo-management" for backwards compatibility. See https://github.com/sourcegraph/sourcegraph/issues/54705
```
Comment (mrnugget) [inline .github/workflows/label-move.yml:256]:
```suggestion
      - name: Get issue if relevant # checking for "source",  but also "iam" and "repo-management" for backwards compatibility. See https://github.com/sourcegraph/sourcegraph/issues/54705
```

**Review by mrnugget — APPROVED** [review]

**Pull request merged**